scripts/main/settings.js

```diff
@@ -562,7 +562,8 @@ settings.save = function (params) {
 		if (data === true) {
 			loadingBar.show("success", lychee.locale["SETTINGS_SUCCESS_UPDATE"]);
 			view.full_settings.init();
-			// lychee.init();
+			// re-read settings
+			lychee.init();
```

Comment:
Uncommenting `lychee.init()` has the (undesired) side-effect that the settings page is closed and the top album page is loaded while leaving the sidebar menu open. The cause for this are 
```
		$(window).bind("popstate", lychee.load);
		lychee.load();
```
in `lychee.js` l.321. It would potentially make sense to add an optional argument `autoload = true` to `lychee.init` so that by default the above two lines are executed but in `settings.js` the call could be changed to `lychee.init(false)` to preserve the original behaviour while updating the lychee settings.

Reply:
I would be in favor of that modification.

Reply:
I was going to suggest to simply hide the sidebar instead, and I think that's perfectly sensible if the use clicks on "Save my modifications ...", but if they instead simply press _Enter_ after changing a single setting, I can see how one may want to stay in the Settings then...

Reply:
I do not have any strong opinions or preferences here. I mainly brought this up because it turns out that uncommenting `lychee.init()` changes the GUI behaviour, which I hadn't realized at first. In fact, I start to like that "Save my modifications..." returns to the album page with the sidebar open because it returns to the layout after clicking settings initially.

However, I am less sure about saving settings using *Enter* and was rather wondering whether this action and modal panel are not a bit misleading. At least it is not entirely obvious what gets saved, i.e. just the entry where *Enter* was pressed or all settings (without checking, I am pretty sure it's the latter but that is not fully transparent to the user).

As a consequence I was thinking whether it wouldn't make sense to retire the *Enter* key with its modal dialog and force the user to scroll down and click "Save my modifications..." once all settings are adjusted. Anyways, just adding yet another option 😉

Reply:
I know for a fact that @ildyria hates the idea of retiring the Enter key there :stuck_out_tongue_winking_eye:.

Reply:
That is to put it mildly.

Reply:
Ok, I'll look into ways that saving with the *Enter* key stays on the settings page. More soon.

Reply:
Done.